server/expression/any.go

```diff
@@ -0,0 +1,359 @@
+		return nil, fmt.Errorf("expected vitess child to be an expression but has type `%T`", children[0])
+	}
+	right, ok := children[1].(sql.Expression)
+	if !ok {
+		return nil, fmt.Errorf("expected vitess child to be an expression but has type `%T`", children[1])
+	}
+	return a.WithChildren(left, right)
+}
+
+// String implements the fmt.Stringer interface.
+func (a *AnyExpr) String() string {
+	if a.leftExpr == nil || a.rightExpr == nil {
+		return fmt.Sprintf("? %s (?)", a.name)
+	}
+	return fmt.Sprintf("%s %s (%s)", a.leftExpr, a.name, a.rightExpr)
+}
+
+// DebugString implements the Expression interface.
+func (a *AnyExpr) DebugString() string {
+	return fmt.Sprintf("%s %s (%s)", sql.DebugString(a.leftExpr), a.name, sql.DebugString(a.rightExpr))
+}
+
+// subqueryWithChildren resolves the comparison functions for a plan.Subquery.
+func subqueryWithChildren(anyExpr *AnyExpr, sub *plan.Subquery) (sql.Expression, error) {
+	schema := sub.Query.Schema()
+	subTypes := make([]pgtypes.DoltgresType, len(schema))
+	for i, col := range schema {
+		dgType, ok := col.Type.(pgtypes.DoltgresType)
+		if !ok {
+			return nil, fmt.Errorf("expected right child to be a DoltgresType but got `%T`", sub)
+		}
+		subTypes[i] = dgType
+	}
+
+	op, err := framework.GetOperatorFromString(anyExpr.subOperator)
+	if err != nil {
+		return nil, err
+	}
+
+	if leftType, ok := anyExpr.leftExpr.Type().(pgtypes.DoltgresType); ok {
+		// Resolve comparison functions once and reuse the functions in Eval.
+		staticLiteral := &Literal{typ: leftType}
+		arrayLiterals := make([]*Literal, len(subTypes))
+		// Each expression may be a different type (which is valid), so we need a comparison function for each expression.
+		compFuncs := make([]*framework.CompiledFunction, len(subTypes))
+		for i, rightType := range subTypes {
+			arrayLiterals[i] = &Literal{typ: rightType}
+			compFuncs[i] = framework.GetBinaryFunction(op).Compile("internal_any_comparison", staticLiteral, arrayLiterals[i])
+			if compFuncs[i] == nil {
+				return nil, fmt.Errorf("operator does not exist: %s = %s", leftType.String(), rightType.String())
+			}
+			if compFuncs[i].Type().(pgtypes.DoltgresType).BaseID() != pgtypes.DoltgresTypeBaseID_Bool {
+				// This should never happen, but this is just to be safe
+				return nil, fmt.Errorf("%T: found equality comparison that does not return a bool", anyExpr)
+			}
+		}
+
+		anyExpr.subqueryAnyExpr = &subqueryAnyExpr{
+			rightSub:      sub,
+			staticLiteral: staticLiteral,
+			arrayLiterals: arrayLiterals,
+			compFuncs:     compFuncs,
+		}
+	}
+
+	return anyExpr, nil
+}
+
+// expressionWithChildren resolves the comparison functions for a sql.Expression.
+func expressionWithChildren(anyExpr *AnyExpr) (sql.Expression, error) {
+	rightType, ok := anyExpr.rightExpr.Type().(pgtypes.DoltgresType)
+	if !ok {
+		return nil, fmt.Errorf("expected right child to be a DoltgresType but got `%T`", anyExpr.rightExpr)
+	}
+
+	if at, ok := rightType.(pgtypes.DoltgresArrayType); ok {
```

Comment:
We could remove this check and bundle it with the one above, so we check for `pgtypes.DoltgresArrayType` instead of `pgtypes.DoltgresType`

Reply:
Also, we should probably error if it's not an array type, which the bundling would take care of